### ISSUE TITLE
Subject: Help with AutoDock-GPU installation – CUDA Verification Failed

### ISSUE THREAD

**Issue by khandokaryogesh** [ccsb-scripps/AutoDock-GPU#275]
Hello,

I am new to molecular docking and currently trying to install and run AutoDock-GPU on my Ubuntu system. While following the installation instructions for AutoDock-GPU, I encountered an issue when running the make DEVICE=CUDA command. Here is the error I get
_**Using Cuda 12.0
Compiling for targets: 52 53 60 61 62 70 72 75 80 86 87 89 90
Makefile:34: *** Cuda verification failed.  Stop.**_

 have installed CUDA 12.6, and I have verified that the necessary CUDA libraries and header files are present in the system. My GPU is an NVIDIA L40S, and nvidia-smi reports the following:

Driver Version: 560.35.03
CUDA Version: 12.6

I have set the following environment variables:

export GPU_INCLUDE_PATH=/usr/local/cuda/include
export GPU_LIBRARY_PATH=/usr/local/cuda/lib64
Despite this setup, the CUDA verification step fails, preventing the compilation of AutoDock-GPU. Could you please guide me on how to resolve this issue or identify what might be going wrong?

Any help would be greatly appreciated!

**Comment by rwxayheee** [comment]
Hi @khandokaryogesh 
If you haven't changed Makefile:
https://github.com/ccsb-scripps/AutoDock-GPU/blob/2e658c318d60806bc6bb361f11f8535d5fb1fe1f/Makefile#L29-L34

Based on the output you got, you were able to locate nvcc (which indicates version 12.0, not 12.6; this is likely a nvcc in your PATH but not the one you wanted) but the target numbers were not printed. It's very likely that the test executable (test_cuda) were not compiled successfully by:
https://github.com/ccsb-scripps/AutoDock-GPU/blob/2e658c318d60806bc6bb361f11f8535d5fb1fe1f/test_cuda.sh#L34
(which is `nvcc -I$GPU_INCLUDE_PATH -L#GPU_LIBRARY_PATH -lcuda -lcudart -o test_cuda test_cuda.cpp`)

I have only tested up to CUDA 12.4. I also never tested the driver version or on NVIDIA L40S. Not very sure what went wrong with the compilation of test codes, but it might be helpful to run `nvcc -I$GPU_INCLUDE_PATH -L$GPU_LIBRARY_PATH -lcuda -lcudart -o test_cuda test_cuda.cpp`, without `&> /dev/null`, to see the standard output and errors. Also, if you want to use CUDA 12.6, make sure you have the right nvcc in PATH.

**Comment by atillack** [comment]
@khandokaryogesh I agree with @rwxayheee, this is likely a case of `nvcc` in your path being different from the directories used. Please post the output of the test compilation `nvcc -I$GPU_INCLUDE_PATH -L$GPU_LIBRARY_PATH -lcuda -lcudart -o test_cuda test_cuda.cpp`.
